modules/_labs/core/react/lib/theming/breakpoints.ts

```diff
@@ -0,0 +1,64 @@
+export enum BreakpointKey {
+  zero = 'zero',
+  s = 's',
+  m = 'm',
+  l = 'l',
+  xl = 'xl',
+}
+
+export type BreakpointFnParam = BreakpointKey | keyof typeof BreakpointKey;
+
+export type CanvasBreakpoints = {
+  zero: number;
+  s: number;
+  m: number;
+  l: number;
+  xl: number;
+};
+
+export const breakpointKeys = ['zero', 's', 'm', 'l', 'xl'] as const;
+
+export const breakpoints: CanvasBreakpoints = {
```

Comment:
This may be a separate discussion but I think a breakpoint system like this might be a bit too stiff. It's assuming that all that matters is `width`. What if someone wants to have more complex [media queries](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries) (height and width for example)?

One alternative is to just to keep this a simple `{[key: string]: string}` object and match breakpoint names to media queries directly:
```
// default breakpoints
{
  ...
  xl: '@media (min-width: 1920px)'
}
```

And then users can overwrite the media queries themselves or add new ones if they need to be more flexible.

Reply:
Hmm that's fair. I believe the idea was that this is for our own components rather than than for consumers to use (though it is available to them if they want it). I figured we'd only have media queries based on width and anything that's height based would be case-specific. So you could do something like ``@media (min-height: 500px) and (min-width: ${theme.breakpoints.l})`` if absolutely necessary.

Since this is going into labs, we have flexibility for now. IMO, this is good enough and we can iterate on it if we feel we need to after getting some feedback.